### PR TITLE
Fix for multiple lines fo the same lighttable module shortcut

### PR DESCRIPTION
**Pull request by Mark-64**
This is a fix for a bug causing multiple entries of the same shortcut in the shortcut list for lighttable modules.

Switching back and forth between lighttable and darkroom made the shortcut list grow continuously.
If Lua module was registered, the list was doubling.

Screenshot

### CODE REVIEW

File: src/gui/accelerators.c [hunk 1]
```diff
@@ -185,7 +185,8 @@ void dt_accel_register_lib_for_views(dt_lib_module_t *self, dt_view_type_flags_t
   accel->local = FALSE;
   // we get the views in which the lib will be displayed
   accel->views = views;
-  darktable.control->accelerator_list = g_slist_prepend(darktable.control->accelerator_list, accel);
+  if (!dt_accel_find_by_path(accel->path))
+    darktable.control->accelerator_list = g_slist_prepend(darktable.control->accelerator_list, accel);
```
Comment (AlicVB) [inline src/gui/accelerators.c]: we need to free accel (or better make the test before the allocation)

Comment (Mark-64) [reply src/gui/accelerators.c]: Checking before allocation as you suggested.
the function is called multiple times, not only when initializing accels, but also when switching views (which make the list grow continuously) and when registering lua modules.
So perhaps there is a better way to fix at higher level, but I don't want to break other things

**Comment by Mark-64** [comment]
In the last commit I put checking before allocation as you suggested.
The function is called multiple times, not only when initializing accels, but also when switching views (which make the list grow continuously) and when registering lua modules.
So perhaps there is a better way to fix at higher level and avoid that the function is even called, but I don't want to break other things

**Comment by Mark-64** [comment]
@AlicVB 
I will continue to dig into the code to see if there are other potential adverse effects of the bug which my patch doesn't cure.  At the moment I don't see any.
As I optimized the patch as you requested, I suggest to accept this PR in the meantime, it's just one line and it can be easily rolled back if a better solution is found.

**Comment by AlicVB** [comment]
Ok, I've found the source of the problem : it's presets !
here : https://github.com/darktable-org/darktable/blob/1fd728123f345588c2d34df757218f6e970a52f7/src/libs/lib.c#L1144
we init _and_ connect each accels, and this called each time you enter a view with the module...
Ideally we should split that in 2 : initialization of presets accels in the module init routine, and connection like for instance !

But well, I don't have the time atm to change and test that, so either someone (you ?) is brave enough to do it, either we go with your current solution (which work but add some useless check)

**Comment by Mark-64** [comment]
Yeah, I found that, but my first attempt to split the function didn't work.
I will try again

**Comment by TurboGit** [comment]
Thanks ! Merging in master and 3.0.1.

**Pull request merged**